istioctl/cmd/analyze.go

```diff
@@ -109,6 +112,9 @@ istioctl analyze -d true a.yaml b.yaml services.yaml
 # Analyze the current live cluster, overriding service discovery to disabled
 istioctl analyze -k -d false
 
+# Analyze the current live cluster and suppress PodMissingProxy for pod mypod in namespace 'testing'.
+istioctl analyze -k --suppress "IST0103=Pod mypod.testing"
```

Comment:
Have the example demo multiple suppressions at once? e.g.

```
--suppress "IST0103=Pod mypod.testing,IST0101=Service myservice.testing"
```

This does highlight the grossness of having comma-separated fields that contain spaces, unfortunately, but it does make the way (or at least *a* way) to suppress multiple things clear.

And if we do include the code-only case in this PR, of course make sure that's also in the example.
```
--suppress "IST0102,IST0103=Pod mypod.testing,IST0101=Service myservice.testing"
```

Reply:
Actually, I tested it, and it turns out commas within the argument are not supported - instead they must be repeated:
`--suppress "IST0103=Pod mypod.testing" --suppress "IST0101=Service myservice.testing"`

(it's slightly more readable this way too, so I'm not sure if I should add support for using commas in a single argument or not

Reply:
A short flag might be useful in this case. kubectl already uses `-s` to specify the kube-apiserver server address. Maybe `-S`?

Reply:
Added support for -S

Reply:
curious why we have format "Pod Foo" instead of k8s format of pod/foo? It's also nice that that doesn't require quoting

Reply:
> curious why we have format "Pod Foo" instead of k8s format of pod/foo? It's also nice that that doesn't require quoting

The args use it to be consistent with the validation message output, which uses ```<kind> <name>.<namespace>``` as the format. Originally the messages used the k8s format of ```<kind>/<namespace>/<name>```, but we changed that to be consistent with other istioctl command outputs. (See https://github.com/istio/istio/issues/17924, https://github.com/istio/istio/pull/18051). Should we revisit that decision? @esnible, do you have any thoughts on this?

Reply:
@selmanj  I'm fine with not dealing with commas and just repeating -S multiple times. We can always add it later if we change our mind. 

Please do add / update the example to show how to do both multiple suppressions and "suppress a code on every resource".

Reply:
Done. PTAL.